language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -479,9 +479,13 @@ public static String getRecordFieldCompletionInsertText(RecordFieldSymbol bField
             insertText.append("{").append(LINE_SEPARATOR);
             List<String> requiredFieldInsertTexts = new ArrayList<>();
             for (RecordFieldSymbol field : requiredFields) {
-                String fieldText = String.join("", Collections.nCopies(tabOffset + 1, "\t")) +
-                        getRecordFieldCompletionInsertText(field, tabOffset + 1);
-                requiredFieldInsertTexts.add(fieldText);
+                // If the field refers to the same type as bField, it results in a stack overflow error.
+                // Avoiding that using the following check
+                if (!bField.equals(field)) {
+                    String fieldText = String.join("", Collections.nCopies(tabOffset + 1, "\t")) +
+                            getRecordFieldCompletionInsertText(field, tabOffset + 1);
+                    requiredFieldInsertTexts.add(fieldText);
+                }
```

Comment:
I believe this check only saves from single level recursions. This is sufficient for the moment.

Reply:
Yes. As discussed via chat, I did update this method to accept a list of parents and a `tabOffset`